osm2gtfs/creators/trips_creator.py

```diff
@@ -13,32 +19,301 @@ def __repr__(self):
         return rep
 
     def add_trips_to_feed(self, feed, data):
-        raise NotImplementedError("Should have implemented this")
+        """
+        This function generates and adds trips to the GTFS feed.
+
+        It is the place where geographic information and schedule is
+        getting joined to produce a routable GTFS.
+        """
+        all_trips_count = 0
+
+        # Go though all lines
+        for line_id, line in data.routes.iteritems():
+
+            print("\nGenerating schedule for line: [" + str(
+                line_id) + "] - " + line.name)
+
+            # Loop through it's itineraries
+            itineraries = line.get_itineraries()
+            for itinerary in itineraries:
+                trips_count = 0
+
+                # Verify data before proceeding
+                if self._verify_data(data.schedule, line, itinerary):
+
+                    # Prepare itinerary's trips and schedule
+                    prepared_trips = self._prepare_trips(feed, data.schedule,
+                                                         itinerary)
+
+                    # Add itinerary shape to feed.
+                    shape_id = self._add_shape_to_feed(
+                        feed, "relation/" + str(itinerary.osm_id), itinerary)
```

Comment:
Coudn't we store this as a field or a method inside `itinerary` instead of hardcoding it here and in many other places?

Reply:
I guess, we want people to be able to override the `shape_id` for their GTFS, as our different network implementations are already using different approaches to these ids. On the `Itinerary` object a `shape_id` is not necessary, because the shape is directly attached to it. This is why I thought it would be good to have an overrideable function in the `trips_creator`.

Reply:
I was referring to the `"relation/" + str(itinerary.osm_id)` part, but I see your point about the IDs.

Reply:
Yes, done, with the latest change to inherit the basic information (also `osm_type` ) from the basic `Element` class.